html/js/product-multilingual.js

```diff
@@ -544,26 +544,33 @@ function initializeTagifyInput(el) {
     });
 
     let abortController;
+    let debounceTimer;
+    const timeoutWait = 300;
+
     input.on("input", function(event) {
         const value = event.detail.value;
         input.whitelist = null; // reset the whitelist
 
         if (el.dataset.autocomplete && el.dataset.autocomplete !== "") {
-            // https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort
-            if (abortController) {
-                abortController.abort();
-            }
+            clearTimeout(debounceTimer);
+
+            debounceTimer = setTimeout(function(){
+                // https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort
+                if (abortController) {
```

Comment:
Shouldn't we do the abort before the setTimeout? i.e. when the user types something, we cancel the previous request immediately, and then we issue the request after 300ms? Otherwise we could get back suggestions that don't match all the letters we typed.